File: 1704. Determine if String Halves Are Alike/halvesAreAlike.java
class Solution {
    public boolean isVowel(char c) {
        if (c == 'a' || c == 'A')
            return true;
        else if (c == 'e' || c == 'E')
            return true;
        else if (c == 'i' || c == 'I')
            return true;
        else if (c == 'o' || c == 'O')
            return true;
        else if (c == 'u' || c == 'U')
            return true;

        return false;
    }

    public boolean halvesAreAlike(String s) {

        int n = s.length();
        int total = 0, cur = 0;
        for (int i = 0; i < n; i++) {

            if (isVowel(s.charAt(i))) {
                total++;
                if (i < n / 2)
                    cur++;
            }
        }

        return 2 * cur == total;
    }
}

File: 1704. Determine if String Halves Are Alike/halvesAreAlike.cpp
class Solution
{
public:
    bool halvesAreAlike(string s)
    {

        int n = s.size();
        int cnt = 0;
        for (int i = 0; i < n / 2; i++)
        {
            if (s[i] == 'a' || s[i] == 'A')
                cnt++;
            else if (s[i] == 'e' || s[i] == 'E')
                cnt++;
            else if (s[i] == 'i' || s[i] == 'I')
                cnt++;
            else if (s[i] == 'o' || s[i] == 'O')
                cnt++;
            else if (s[i] == 'u' || s[i] == 'U')
                cnt++;
        }
        for (int i = n / 2; i < n; i++)
        {
            if (s[i] == 'a' || s[i] == 'A')
                cnt--;
            else if (s[i] == 'e' || s[i] == 'E')
                cnt--;
            else if (s[i] == 'i' || s[i] == 'I')
                cnt--;
            else if (s[i] == 'o' || s[i] == 'O')
                cnt--;
            else if (s[i] == 'u' || s[i] == 'U')
                cnt--;
        }

        return cnt == 0 ? true : false;
    }
};